packages/react-vanilla-components/src/components/Switch.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useCallback } from 'react';
+import { withRuleEngine } from '../utils/withRuleEngine';
+import { PROPS } from '../utils/type';
+import FieldWrapper from './common/FieldWrapper';
+
+const Switch = (props: PROPS) => {
+  const {
+    id,
+    label,
+    enum: enums,
+    enumNames,
+    required,
+    value,
+    name,
+    enabled,
+    visible,
+    appliedCssClassNames
+  } = props;
+
+  const selectedValue = enums?.[0];
+  const handleChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const val = e.target.checked ? selectedValue : 'undefined';
+      props.dispatchChange(val);
+    },
+    [props.dispatchChange]
+  );
+  return (
+    <div
+      className={`cmp-adaptiveform-switch cmp-adaptiveform-switch--${
+        selectedValue === value ? 'filled' : 'empty'
+      } ${appliedCssClassNames || ''}`}
+      data-cmp-is="adaptiveFormSwitch"
+      data-cmp-visible={visible}
+      data-cmp-enabled={enabled}
+      data-cmp-required={required}
+      id={id}
+    >
+      <FieldWrapper
+        bemBlock="cmp-adaptiveform-switch"
+        label={label}
+        id={id}
+        tooltip={props.tooltip}
+        description={props.description}
+        isError={props.isError}
+        errorMessage={props.errorMessage}
+      >
+        <div className="cmp-adaptiveform-switch__container">
+          <span className="cmp-adaptiveform-switch__option cmp-adaptiveform-switch__option--off"></span>
```

Comment:
not align with core component. 
Why are you not showing off text

Reply:
Applied the logic!